python_bindings/bindings.cpp

```diff
@@ -0,0 +1,1144 @@
+        .def(py::init<>())
+        .def(py::init<Seldon::SimpleAgentData>())
+        .def_readwrite("data", &Seldon::Agent<Seldon::SimpleAgentData>::data);
+
+    //------------------------------------------------------------------------------------------------------------------------------------------
+
+    py::class_<Seldon::DiscreteVectorAgentData>(m, "DiscreteVectorAgentData")
+        .def(py::init<>())
+        .def_readwrite("opinion", &Seldon::DiscreteVectorAgentData::opinion);
+
+    py::class_<Seldon::Agent<Seldon::DiscreteVectorAgentData>>(m, "DiscreteVectorAgent")
+        .def(py::init<>())
+        .def(py::init<Seldon::DiscreteVectorAgentData>())
+        .def_readwrite("data", &Seldon::Agent<Seldon::DiscreteVectorAgentData>::data);
+
+    //------------------------------------------------------------------------------------------------------------------------------------------
+
+    py::class_<Seldon::ActivityAgentData>(m, "ActivityAgentData")
+        .def(py::init<>())
+        .def_readwrite("opinion", &Seldon::ActivityAgentData::opinion)
+        .def_readwrite("activity", &Seldon::ActivityAgentData::activity)
+        .def_readwrite("reluctance", &Seldon::ActivityAgentData::reluctance);
+
+    py::class_<Seldon::Agent<Seldon::ActivityAgentData>>(m, "ActivityAgent")
+        .def(py::init<>())
+        .def(py::init<Seldon::ActivityAgentData>())
+        .def_readwrite("data", &Seldon::Agent<Seldon::ActivityAgentData>::data);
+
+    //------------------------------------------------------------------------------------------------------------------------------------------
+
+    py::class_<Seldon::InertialAgentData>(m, "InertialAgentData")
+        .def(py::init<>())
+        .def_readwrite("opinion", &Seldon::InertialAgentData::opinion)
+        .def_readwrite("activity", &Seldon::InertialAgentData::activity)
+        .def_readwrite("reluctance", &Seldon::InertialAgentData::reluctance)
+        .def_readwrite("velocity", &Seldon::InertialAgentData::velocity);
+
+    py::class_<Seldon::Agent<Seldon::InertialAgentData>>(m, "InertialAgent")
+        .def(py::init<>())
+        .def(py::init<Seldon::InertialAgentData>())
+        .def_readwrite("data", &Seldon::Agent<Seldon::InertialAgentData>::data);
+
+    //------------------------------------------------------------------------------------------------------------------------------------------
+
+    py::class_<Seldon::Simulation<Seldon::SimpleAgent>>(m, "SimulationSimpleAgent")
+        .def(py::init<>([](const Seldon::Config::SimulationOptions &options,
+                           const std::optional<std::string> &agent_file,
+                           const std::optional<std::string> &network_file) {
+                 Seldon::Config::validate_settings(options);
+                 Seldon::Config::print_settings(options);
+                 return new Seldon::Simulation<Seldon::SimpleAgent>(options, network_file, agent_file);
```

Comment:
Why `return new`? Run `valgrind` perhaps. It would be cleaner here to have a WrapperSimulationSimpleAgent which has a pointer to Seldon::Simulation and then have the pointer deleted in its destructor.

Reply:
Also for the classes below